### PR TITLE
MAINT/DOC: `special.nrdtrimn/nrdtrisd` docstring fixes

### PR DESCRIPTION
**Pull request by dschmitz89**
#### Reference issue
Related to #19953 and #7168

#### What does this implement/fix?

1. Fixes the docstring of these functions that were in the wrong order (see [here](https://github.com/scipy/scipy/blob/913c3ffafd1179dccc723a23fca3f0d661f633dc/scipy/special/_cdflib_wrappers.pxd#L539))
2. Adds examples to the docstring

### CODE REVIEW

**Comment by ilayn** [comment]
Nice. By the way, please don't forget the possibility that I might have screwed up the argument order while writing the wrappers :) I paid attention trying not to touch anything on the scipy side but who knows. But then again it would break more stuff.

**Comment by dschmitz89** [comment]
No worries about the rewrite, the docstring was apparently broken for many years. The example returns the same output using scipy 1.10.1. Good that the rewrite forces us to look into these functions more ;)

**Review by ilayn — APPROVED** [review]

**Comment by ilayn** [comment]
FFT errors are not unrelated I think.

Could you also remove the entries in line 13, 14 if the tests are added?

**Comment by dschmitz89** [comment]
The test still fails: https://github.com/scipy/scipy/actions/runs/7860753553/job/21448286231?pr=20069

It only shows up when the full test suite is run as the cdflib tests are marked as slow.

I do not understand the origin of the test failure. For some reason,the machinery picks up the wrong parameters to test against each other.

**Comment by ilayn** [comment]
I wish I can understand the testing mechanism there but it is a bit too much of dependency injection for me. I feel like this is about the `_assert_inverts` but can't figure it out.

**Comment by dschmitz89** [comment]
> I wish I can understand the testing mechanism there but it is a bit too much of dependency injection for me. I feel like this is about the `_assert_inverts` but can't figure it out due to my lack of mpmath and special knowledge .

Could not agree more. It looks easy to use in principle even without special function expertise but if something fails, I find it hard to debug.

File: scipy/special/_add_newdocs.py [hunk 1]
```diff
@@ -11091,7 +11091,31 @@ def add_newdoc(name, doc):
...
+    >>> from scipy.stats import norm
+    >>> mean = 3.
+    >>> std = 2.
+    >>> x = 6.
+    >>> p = norm.cdf(x, loc=mean, scale=std)
+    >>> p
+    0.9331927987311419
+
+    Verify that `nrdtrimn` returns the original value for `mean`.
+    >>> from scipy.special import nrdtrimn
```
Comment (ilayn) [inline scipy/special/_add_newdocs.py:11116]: this one and the other example requires newlines after text apparently 

https://output.circle-artifacts.com/output/job/ed62f044-399b-4392-9f29-061f67c3847a/artifacts/0/html/reference/generated/scipy.special.nrdtrimn.html#scipy.special.nrdtrimn

File: scipy/special/_add_newdocs.py [hunk 2]
```diff
@@ -11119,7 +11144,31 @@ def add_newdoc(name, doc):
...
+    >>> from scipy.stats import norm
+    >>> mean = 3.
+    >>> std = 2.
+    >>> x = 6.
+    >>> p = norm.cdf(x, loc=mean, scale=std)
+    >>> p
+    0.9331927987311419
+
+    Verify that `nrdtrisd` returns the original value for `std`.
+    >>> from scipy.special import nrdtrisd
```
Comment (ilayn) [inline scipy/special/_add_newdocs.py:11169]: https://output.circle-artifacts.com/output/job/ed62f044-399b-4392-9f29-061f67c3847a/artifacts/0/html/reference/generated/scipy.special.nrdtrisd.html#scipy.special.nrdtrisd

same here

**Comment by dschmitz89** [comment]
Sorry for that rookie mistake. Looks good now. I wish the linter/refguide check would catch that (would numpydoc validation help?).

**Comment by lucascolley** [comment]
> would numpydoc validation help?

I don't think so based on https://numpydoc.readthedocs.io/en/latest/validation.html#built-in-validation-checks

File: scipy/special/_add_newdocs.py [hunk 3]
```diff
@@ -11103,12 +11128,13 @@ def add_newdoc(name, doc):
...
     Parameters
     ----------
+    mn : scalar or ndarray
+        The mean of the normal distribution.
     p : array_like
         CDF values, in range (0, 1].
     x : array_like
         Quantiles, i.e. the upper limit of integration.
-    mn : scalar or ndarray
-        The mean of the normal distribution.
```
Comment (steppi) [inline scipy/special/_add_newdocs.py:11111]: The signature above says `nrdtrisd(p, x, mn, out=None)`, but here the order is `mn, `p`, `x`. I think these should be consistent.

Comment (dschmitz89) [reply scipy/special/_add_newdocs.py:11111]: 🤦 done

**Review by steppi — APPROVED** [review]
Looks good now. Also, I noticed that `nrdtrisd` returns garbage for infeasible inputs, say when you have a CDF value > 0.5 but x is less than the mean. e.g `nrdtrisd(0, 0.6, -1) = -3.9471538755427473`. This is because `nrdtrisd` is really finding a scale parameter, not a standard deviation, and a negative scale gives a reflection of the original distribution. This should be addressed at some point too in a future PR. Either by changing the behavior, or documenting the existing behavior.

**Comment by steppi** [comment]
Thanks @dschmitz89!

**Pull request merged**